### DISCUSSION TITLE
How to define "at least one property required"?

### DISCUSSION THREAD

**Discussion by davideroffo** [sinclairzx81/typebox#518] [Q&A] [answered]
Good evening, everyone :city_sunset:
I have a question regarding fastify schemas using `@sinclair/typebox`:
I'm trying to create a schema where my endpoint triggers a validation error if none of the possible body parameters are passed. 
Consider the following example:
```
const Schema = Type.Partial(
  Type.Object({
    a: Type.String(),
    b: Type.String(),
    c: Type.String()
  })
);
```
Basically, I want to ensure that at least one of the parameters `a`, `b`, or `c` is required. How can I achieve this?
Thank you :pray:

**Comment by sinclairzx81** [comment]
@davideroffo Hi,

Try `minProperties: 1`

```typescript
const Schema = Type.Partial(
  Type.Object({
    a: Type.String(),
    b: Type.String(),
    c: Type.String()
  })
, { minProperties: 1 })

// tests
//
import { Value } from '@sinclair/typebox/value'

Value.Check(Schema, { a: '', b: '', c: '' }) // true
Value.Check(Schema, { a: '', b: '' })        // true
Value.Check(Schema, { a: '' })               // true
Value.Check(Schema, { b: '' })               // true
Value.Check(Schema, { c: '' })               // true
Value.Check(Schema, {})                      // false
```

**Reply by davideroffo** [reply]
Thanks a lot for your quick reply.
I just tried, but the problem is that if I pass a `d` parameter is not throwing any validation error:

```javascript
const Schema = Type.Partial(
  Type.Object({
    a: Type.String(),
    b: Type.String(),
    c: Type.String()
  })
, { minProperties: 1 })

Value.Check(Schema, { d: '' })               // true
```

I want it to raise a validation error even if I'm passing properties that shouldn't be passed.

**Reply by davideroffo** [reply, marked as answer]
I solved it by using `minProperties: 1` in conjunction with `additionalProperties: false`, following this Stack Overflow answer: https://stackoverflow.com/a/75233932/5830331 .

```javascript
const server = Fastify({
  ajv: {
    customOptions: {
      removeAdditional: false
    }
  }
})

const Schema = Type.Partial(
  Type.Object({
    a: Type.String(),
    b: Type.String(),
    c: Type.String()
  })
, { minProperties: 1, additionalProperties: false })

Value.Check(Schema, { d: '' })               // false
```
